spark/src/main/java/org/opensearch/sql/spark/execution/statestore/StatementStorageService.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.spark.execution.statestore;
+
+import java.util.Optional;
+import org.opensearch.sql.spark.execution.statement.StatementModel;
+import org.opensearch.sql.spark.execution.statement.StatementState;
+
+public interface StatementStorageService {
+
+  StatementModel createStatement(StatementModel statementModel, String datasourceName);
```

Comment:
can we make datasourceName as part of StatementModel?

Reply:
That's a good point. Each data model include datasourceName as field, and we can refer that instead of the parameter. We cannot remove for getXxx parameter. I'll change that in separate PR since it require change for all the storage services.